src/providers/ModalsProvider.tsx

```diff
@@ -0,0 +1,63 @@
+import React, {
+  PropsWithChildren,
+  createContext,
+  useCallback,
+  useState,
+} from "react"
+import styles from "./ModalProvider.module.scss"
+
+interface ModalsContext {
+  content?: React.ReactNode
+  isOpen?: boolean
+  onPresent: (content: React.ReactNode, key?: string) => void
+  onClose: () => void
+}
+
+export const Context = createContext<ModalsContext>({
+  onPresent: () => undefined,
+  onClose: () => undefined,
```

Comment:
`() => void` describes this action more precisely since technically you aren't returning undefined but rather nothing is happening, hence void. Also leverages TS here.

Reply:
Right, I have tried to give void, but it shows me type error. That is the reason why I gave undefined instead of void 
![Screen Shot 2021-12-10 at 2 02 58 PM](https://user-images.githubusercontent.com/75422800/145564097-5a3d7602-defe-4dbc-a539-059e8ac97aff.png)

Reply:
I mis-read. This should be `() => {}` which more closely represent a `noop`